src/core/util/stringconversion.cpp

```diff
@@ -60,9 +60,20 @@ std::wstring toWstring(const std::string& str) {
 #else
     auto state = std::mbstate_t();
     auto sptr = str.data();
+    const char* loc = nullptr;
     size_t len = std::mbsrtowcs(nullptr, &sptr, 0, &state);
+    if (len == static_cast<std::size_t>(-1)) {
+        loc = std::setlocale(LC_CTYPE, nullptr);
+        std::setlocale(LC_CTYPE, "en_US.UTF-8");
+        len = std::mbsrtowcs(nullptr, &sptr, 0, &state);
+        if (len == static_cast<std::size_t>(-1)) {
+            if (loc) std::setlocale(LC_CTYPE, loc);
```

Comment:
Should be an else here for throwing the exception?

Reply:
no, I just want to restore the local before I leave by throwing